src/components/select/__internal__/utils/highlight-part-of-text.tsx

```diff
@@ -14,9 +14,12 @@ function highlightPartOfText(
     return text;
   }
 
-  const precedingText = text.substr(0, indexOfFirstMatch);
-  const matchingText = text.substr(indexOfFirstMatch, partToHighlight.length);
-  const followingText = text.substr(
```

Comment:
`substr` is a deprecated method for strings, so replaced with `slice`

Reply:
Comment (non-blocking): I wonder if using [substring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#the_difference_between_substring_and_substr) here would be a little more intuitive, not a huge deal either way though as long as we can guarantee the slight differences between [substring and slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#differences_between_substring_and_slice) aren't an issue for us